Source/Core/DolphinQt/RenderWidget.cpp

```diff
@@ -80,6 +87,35 @@ void RenderWidget::SetFillBackground(bool fill)
   setAutoFillBackground(fill);
 }
 
+void RenderWidget::dragEnterEvent(QDragEnterEvent* event)
+{
+  if (event->mimeData()->hasUrls() && event->mimeData()->urls().size() == 1)
+    event->acceptProposedAction();
+}
+
+void RenderWidget::dropEvent(QDropEvent* event)
+{
+  const auto& urls = event->mimeData()->urls();
+  if (urls.empty())
+    return;
+
+  const auto& url = urls[0];
+  QFileInfo file_info(url.toLocalFile());
+
+  auto path = file_info.filePath();
+
+  if (!file_info.exists() || !file_info.isReadable())
+  {
+    QMessageBox::critical(this, tr("Error"), tr("Failed to open '%1'").arg(path));
+    return;
+  }
+
+  if (file_info.isFile())
```

Comment:
This can be inverted so it's not structuring the valid codepath like the invalid ones that precede it.

Reply:
This code was lifted from MainWindow.cpp. I can invert it here. Want me to do a pull inverting it in master after?